src/SumoCoders/FrameworkUserBundle/Controller/EditController.php

```diff
@@ -78,6 +78,12 @@ public function editAction(Request $request, ?int $id)
             throw new AccessDeniedHttpException('Access denied.');
         }
 
-        return parent::baseAction($request, $id);
+        $response = parent::baseAction($request, $id);
+
+        if ($response instanceof RedirectResponse && !$this->authorizationChecker->isGranted('ROLE_ADMIN')) {
+            return RedirectResponse::create($request->getPathInfo());
```

Comment:
So, they get redirected to the current page they are viewing?

Reply:
yes